packages/gatsby-theme-store/src/components/SearchPage/index.tsx

```diff
@@ -0,0 +1,74 @@
+import { useLocation } from '@reach/router'
+import { SuspenseViewport } from '@vtex/store-ui'
+import React, { lazy } from 'react'
+import type { FC } from 'react'
+import type { SearchParamsState } from '@vtex/store-sdk'
+
+import { usePixelSendEvent } from '../../sdk/pixel/usePixelSendEvent'
+import { SearchProvider } from '../../sdk/search/Provider'
+import AboveTheFold from './AboveTheFold'
+import BelowTheFoldPreview from './BelowTheFoldPreview'
+import SEO from './SEO'
+import type { ServerSearchPageQueryQuery } from '../../templates/__generated__/ServerSearchPageQuery.graphql'
+import type { BrowserSearchPageQueryQuery } from '../../templates/__generated__/BrowserSearchPageQuery.graphql'
+
+const belowTheFoldPreloader = () => import('./BelowTheFold')
+const BelowTheFold = lazy(belowTheFoldPreloader)
+
+export interface SearchViewProps {
+  pageContext: {
+    canonicalPath: string
+  }
+  data: ServerSearchPageQueryQuery | BrowserSearchPageQueryQuery
+  searchParams: SearchParamsState
+}
```

Comment:
`pageContext` and `data` are attributes that can be extended of `PageProps` from `gatsby`

Reply:
`PageProps` require many other params that are not really passed in `search.browser.tsx` or `search.server.tsx`. Adding this type would result in bad tsc compilation